source/extensions/stat_sinks/metrics_service/grpc_metrics_service_impl.cc

```diff
@@ -39,17 +46,45 @@ void GrpcMetricsStreamerImpl::send(envoy::service::metrics::v3::StreamMetricsMes
   }
 }
 
-MetricsServiceSink::MetricsServiceSink(const GrpcMetricsStreamerSharedPtr& grpc_metrics_streamer,
-                                       const bool report_counters_as_deltas)
-    : grpc_metrics_streamer_(grpc_metrics_streamer),
-      report_counters_as_deltas_(report_counters_as_deltas) {}
-
-void MetricsServiceSink::flushCounter(
-    const Stats::MetricSnapshot::CounterSnapshot& counter_snapshot, int64_t snapshot_time_ms) {
-  io::prometheus::client::MetricFamily* metrics_family = message_.add_envoy_metrics();
-  metrics_family->set_type(io::prometheus::client::MetricType::COUNTER);
-  metrics_family->set_name(counter_snapshot.counter_.get().name());
-  auto* metric = metrics_family->add_metric();
+MetricsPtr MetricsFlusher::flush(Stats::MetricSnapshot& snapshot) const {
+  auto metrics =
+      std::make_unique<Envoy::Protobuf::RepeatedPtrField<io::prometheus::client::MetricFamily>>();
```

Comment:
Are you going to need to override this also? Or if you do will you override the entire flusher?

Reply:
If we overrode, then yes, we would have to override the entire flusher. This proto struct is the minimal container that the flusher uses to aggregate metrics, so I don't see us changing it; unless we wanted to change the metrics format we use on the wire, and in the server-side service. From the core use in Envoy Mobile, and from my conversations with @jingwei99, I haven't come up with a good use case for changing the data format.